src/vellum/workflows/nodes/mocks.py

```diff
@@ -13,5 +105,137 @@ class MockNodeExecution(UniversalBaseModel):
+    def validate_all(
+        raw_mock_workflow_node_configs: List[Any],
+        workflow: Type["BaseWorkflow"],
+    ) -> List["MockNodeExecution"]:
+        ArrayVellumValue.model_rebuild()
+        try:
+            mock_workflow_node_configs = [
+                _RawMockWorkflowNodeConfig.model_validate(raw_mock_workflow_node_config)
+                for raw_mock_workflow_node_config in raw_mock_workflow_node_configs
+            ]
+        except ValidationError as e:
+            raise WorkflowInitializationException(
+                message="Failed to validate mock node executions",
+                code=WorkflowErrorCode.INVALID_INPUTS,
+            ) from e
+
+        nodes = {node.__id__: node for node in workflow.get_nodes()}
+        node_output_name_by_id = {
+            output.__id__: output.name for node in workflow.get_nodes() for output in node.Outputs
+        }
+
+        # We need to support the old way that the Vellum App's WorkflowRunner used to define Node Mocks in order to
+        # avoid needing to update the mock resolution strategy that it and the frontend uses. The path towards
+        # cleaning this up will go as follows:
+        # 1. Release Mock support in SDK-Enabled Workflows
+        # 2. Deprecate Mock support in non-SDK enabled Workflows, encouraging users to migrate to SDK-enabled Workflows
+        # 3. Remove the old mock resolution strategy
+        # 4. Update this SDK to handle the new mock resolution strategy with WorkflowValueDescriptors
+        # 5. Cutover the Vellum App to the new mock resolution strategy
+        # 6. Remove the old mock resolution strategy from this SDK
+        def _translate_raw_logical_expression(
+            raw_logical_expression: _RawLogicalExpression,
+            raw_variables: List[_RawMockWorkflowNodeExecutionValuePointer],
+        ) -> BaseDescriptor:
+            if raw_logical_expression.type == "LOGICAL_CONDITION":
+                return _translate_raw_logical_condition(raw_logical_expression, raw_variables)
+            else:
+                return _translate_raw_logical_condition_group(raw_logical_expression, raw_variables)
+
+        def _translate_raw_logical_condition_group(
+            raw_logical_condition_group: _RawLogicalConditionGroup,
+            raw_variables: List[_RawMockWorkflowNodeExecutionValuePointer],
+        ) -> BaseDescriptor:
+            if not raw_logical_condition_group.conditions:
+                return ConstantValueReference(True)
+
+            conditions = [
+                _translate_raw_logical_expression(condition, raw_variables)
+                for condition in raw_logical_condition_group.conditions
+            ]
+            return reduce(
+                lambda acc, condition: (
+                    acc and condition if raw_logical_condition_group.combinator == "AND" else acc or condition
+                ),
+                conditions,
+            )
+
+        def _translate_raw_logical_condition(
```

Comment:
we don't need to support all logical condition operators bc the app currently only uses execution count >= 0 for now (and for the last year)